themes-default/slim/views/vue-components/qualityChooser.mako

```diff
@@ -0,0 +1,199 @@
+                    <h5>Preferred</h5>
+                    <select v-model="selectedPreffered" name="preferred_qualities" multiple="multiple" :size="preferredQualityList.length" class="form-control form-control-inline input-sm">
+                        <option v-for="quality in preferredQualityList" :selected="quality in preferredQualities" :value="quality">{{qualityStrings[quality]}}</option>
+                    </select>
+                </div>
+            </div>
+            <div id="qualityExplanation">
+                <h5><b>Quality setting explanation:</b></h5>
+                <h5 v-if="selectedPreffered.length === 0">This will download <b>any</b> of these qualities and then stops searching: <label id="allowedExplanation">{{allowedExplanation.join(', ')}}</label></h5>
+                <template v-else>
+                <h5>Downloads <b>any</b> of these qualities: <label id="allowedPreferredExplanation">{{allowedExplanation.join(', ')}}</label></h5>
+                <h5>But it will stop searching when one of these is downloaded:  <label id="preferredExplanation">{{preferredExplanation.join(', ')}}</label></h5>
+                </template>
+            </div>
+            <div v-f="seriesSlug">
+                <h5 class="red-text" id="backloggedEpisodes" v-html="backloggedEpisodes"></h5>
+            </div>
+            <div id="archive" v-if="archive">
+                <h5>
+                    <b>
+                        Archive downloaded episodes that are not currently in <a target="_blank" href="manage/backlogOverview/"><font color="blue"><u>backlog</u>.</font></a>
+                    </b>
+                        <br />Avoids unnecessarily increasing your backlog
+                    </br>
+                </h5>
+                <button @click="archiveEpisodes" class="btn btn-inline" type="button" value="Archive episodes">
+                <h5>{{archivedStatus}}</h5>
+            </div>
+        </div>
+    </div>
+</script>
+<%!
+    import json
+    from medusa import app
+    from medusa.numdict import NumDict
+    from medusa.common import Quality, qualityPresets, qualityPresetStrings
+%>
+<%
+if not show is UNDEFINED:
+    __quality = int(show.quality)
+else:
+    __quality = int(app.QUALITY_DEFAULT)
+allowed_qualities, preferred_qualities = Quality.split_quality(__quality)
+overall_quality = Quality.combine_qualities(allowed_qualities, preferred_qualities)
+selected = None
+%>
+<%
+def convert(obj):
+    ## This converts the keys to strings as keys can't be ints
+    if isinstance(obj, (NumDict, dict)):
+        new_obj = {}
+        for key in obj:
+            new_obj[str(key)] = obj[key]
+        obj = new_obj
+
+    return json.dumps(obj)
+
+%>
+<script>
+Vue.component('quality-chooser', {
+    name: 'quality-chooser',
+    template: '#quality-chooser-template',
+    data() {
+        return {
+            // Python convertions
+            allowedQualities: ${convert(allowed_qualities)},
```

Comment:
Can we change this so it's using props/events and not directly the python vars?

Reply:
For now let's leave this here since otherwise each file that uses this will need that python code which will just duplicate it.

Reply:
In this case I don't fully agree. But it's up to you. And what works best for you. So i'm fine with it.
Besides this, you think it's ready to merge? Needs some more reviewing? When this is ready I can merge it into my editShow pr.

Reply:
Also it'l need a rebase.

Reply:
I think after a rebase it'll still need a little bit of testing as I believe it still has the issue of getting the wrong quality when you first open the edit show page.

Reply:
OK I'll merge it into my branch. Let's test there.